Source/Base/Base/Math/Spline.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include "Base/Types.h"
+
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace Spline
+{
+    struct InterpolatedStorage
+    {
+    protected:
+        bool _dirty = true;
+        u32 _stepSize = 0;
+        std::vector<vec3> _points = {};
+
+    public:
+        explicit InterpolatedStorage(u32 step);
+
+        void SetStep(u32 newStep);
+        void StoreInterpolation(const std::vector<vec3>& points, u32 step);
+        void AddPortion(const std::vector<vec3>& points, i32 position = -1);
+
+        void Clear() { _points.clear(); };
+        void MarkAsDirty() { _dirty = true; };
+        bool CheckSize() { return (_points.size() % _stepSize) == 0; };
+
+        [[nodiscard]] bool IsDirty() { return _dirty; };
+        [[nodiscard]] const std::vector<vec3>& GetInterpolatedStorage(u32 splinePoints = 0);
+    };
+
+    struct SplinePoint
+    {
+        SplinePoint() { };
```

Comment:
We always define if a struct's default scope is public/private, please add it to all structs/classes you've made as part of this PR.